### ISSUE TITLE
Question about  MemoryEfficientSwish and Swish function.

### ISSUE THREAD

**Issue by quangtn266** [zylo117/Yet-Another-EfficientDet-Pytorch#335]
Thank you for your work, but I have one concern about "MemoryEfficientSwish and Swish". Can you explain them ?

**Comment by zylo117** [comment]
memory efficient swish will store the x for backward, instead of calculating it again.

**Comment by quangtn266** [comment]
> memory efficient swish will store the x for backward, instead of calculating it again.

Sorry for the inconvenience, i have another question which isn't related to your repo, just about my personal issue. I don't understand why you and other guy repos use below the code

```
        self.p6_w1 = nn.Parameter(torch.ones(2, dtype=torch.float32), requires_grad=True)
        self.p6_w1_relu = nn.ReLU()
```
What does this mean about these code ? what is difference between using those code and without them ?

**Comment by zylo117** [comment]
Not sure what exactly are you talking about. Can you say it more specifically?

**Comment by quangtn266** [comment]
> Not sure what exactly are you talking about. Can you say it more specifically?

I mean that the above code is used for the below code for calculating p6_up:
`        p6_w1 = self.p6_w1_relu(self.p6_w1)
        weight = p6_w1 / (torch.sum(p6_w1, dim=0) + self.epsilon)
        # Connections for P6_0 and P7_0 to P6_1 respectively
        p6_up = self.conv6_up(self.swish(weight[0] * p6_in + weight[1] * self.p6_upsample(p7_in)))`
So I wonder Can I only use like like:
`
p6_up = self.conv6_up(self.swish( p6_in + self.p6_upsample(p7_in)))
`
I think that both methods are the same, but why we used the first method. What the difference between both of them?

**Comment by zylo117** [comment]
well, then where is p6_w1?

**Comment by quangtn266** [comment]
> well, then where is p6_w1?

Oh, It is my mistake for writing my reply, I already update that. Sorry :D
